bug: daemon end-to-end tests failing - health check timeout after 20s

## Problem

Two daemon end-to-end tests are failing with a 20-second health check timeout:

```
FAILED tests/integration/test_daemon.py::TestEndToEnd::test_daemon_start_stop
FAILED tests/integration/test_daemon.py::TestEndToEnd::test_daemon_embedding_workflow
```

**Error message:**
```
RuntimeError: Failed to start daemon: Daemon process started but not responding 
to health checks after 20s. This may indicate model loading issues. 
Check daemon logs at: /private/.../test-daemon.log
```

## Context

The daemon tests are marked `@pytest.mark.slow` and run with `-m slow`. When run, they fail consistently:
- 49 slow tests pass
- 2 slow tests fail (both daemon end-to-end tests)

The failure occurs in `DaemonLifecycle.start()` where the health check loop times out after 20 seconds waiting for the daemon to respond.

**Possible causes:**
1. Model loading taking longer than 20 seconds on first run
2. Health check endpoint not responding correctly
3. Socket communication issue between test and daemon
4. Daemon process failing silently without proper error reporting

## Code Reference

`tests/integration/test_daemon.py::TestEndToEnd`

Related code:
- `ember/adapters/daemon/lifecycle.py:100-220` (start method with health checks)
- `ember/adapters/daemon/server.py` (daemon server)
- `ember/adapters/daemon/client.py` (health check logic)

## Proposed Investigation

1. Check daemon log file for errors during test
2. Verify health check endpoint is responding
3. Consider increasing timeout for CI environments
4. Add more detailed logging during startup sequence
5. Check if model is being loaded correctly in test environment

## Success Criteria

- [ ] Both daemon end-to-end tests pass
- [ ] Root cause identified and documented
- [ ] Timeout handling improved if needed
- [ ] Error messages provide actionable debugging info

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - Helps users resolve index corruption or stale data issues
 
 ### Fixed
+- **Fix daemon end-to-end tests failing with health check timeout** (#185)
+  - Fixed socket path length issue: Unix domain sockets have ~104 char limit on macOS
+  - Test fixtures now use short paths in `/tmp` instead of pytest's long temp paths
+  - Fixed zombie process detection in `is_process_alive()` by reaping child processes
+  - Increased SIGKILL verification timeout from 0.5s to 2.5s with retry loop
+  - Added `auto_start=False` to fallback test to ensure proper fallback behavior
+
 - **Fix race condition in daemon PID file management** (#152)
   - PID file is now written immediately after spawning daemon process
   - Eliminates race where process could die between alive-check and PID write

--- a/ember/adapters/daemon/lifecycle.py
+++ b/ember/adapters/daemon/lifecycle.py
@@ -73,9 +73,21 @@ class DaemonLifecycle:
             pid: Process ID
 
         Returns:
-            True if process exists
+            True if process exists and is not a zombie
         """
         try:
+            # First try to collect any zombie children to avoid false positives
+            # This handles the case where we spawned a daemon that has exited
+            # but hasn't been reaped yet (zombie state)
+            try:
+                os.waitpid(pid, os.WNOHANG)
+            except ChildProcessError:
+                # Process is not our child, that's fine
+                pass
+            except OSError:
+                # Process doesn't exist or other error
+                pass
+
             # Send signal 0 (no-op, just checks if process exists)
             os.kill(pid, 0)
             return True
@@ -276,14 +288,16 @@ class DaemonLifecycle:
             return False
 
         # Verify SIGKILL worked before cleanup
-        time.sleep(0.5)
-        if not self.is_process_alive(pid):
-            logger.info("Daemon force-killed")
-            self.cleanup_stale_files()
-            return True
-        else:
-            logger.error("Daemon survived SIGKILL! Manual cleanup required.")
-            return False
+        # Give process time to fully terminate (may take longer on some systems)
+        for _ in range(5):  # Up to 2.5s total
+            time.sleep(0.5)
+            if not self.is_process_alive(pid):
+                logger.info("Daemon force-killed")
+                self.cleanup_stale_files()
+                return True
+
+        logger.error("Daemon survived SIGKILL! Manual cleanup required.")
+        return False
 
     def restart(self, foreground: bool = False) -> bool:
         """Restart the daemon.

--- a/tests/integration/test_daemon.py
+++ b/tests/integration/test_daemon.py
@@ -5,6 +5,8 @@ Tests the daemon server, client, and lifecycle management.
 
 
 import signal
+import tempfile
+from pathlib import Path
 from unittest.mock import MagicMock, patch
 
 import pytest
@@ -17,8 +19,20 @@ from ember.adapters.daemon.server import DaemonServer
 
 @pytest.fixture
 def temp_socket_path(tmp_path):
-    """Temporary socket path for testing."""
-    return tmp_path / "test-daemon.sock"
+    """Temporary socket path for testing.
+
+    Uses a short base path (/tmp) to avoid AF_UNIX path length limits (~104 chars on macOS).
+    The pytest tmp_path can be too long for Unix domain sockets.
+    """
+    # Create a short temp directory for the socket
+    # Unix sockets have ~104 char limit on macOS, pytest tmp_path can exceed this
+    short_tmp = tempfile.mkdtemp(prefix="emb_", dir="/tmp")
+    socket_path = Path(short_tmp) / "d.sock"  # Keep filename short too
+    yield socket_path
+    # Cleanup
+    if socket_path.exists():
+        socket_path.unlink()
+    Path(short_tmp).rmdir()
 
 
 @pytest.fixture
@@ -644,8 +658,11 @@ class TestEndToEnd:
     @pytest.mark.slow
     def test_client_fallback_on_daemon_failure(self, temp_socket_path):
         """Test client falls back to direct mode when daemon unavailable."""
-        # Create client with fallback enabled
-        client = DaemonEmbedderClient(socket_path=temp_socket_path, fallback=True)
+        # Create client with fallback enabled but auto_start disabled
+        # This ensures the daemon won't be started, forcing fallback
+        client = DaemonEmbedderClient(
+            socket_path=temp_socket_path, fallback=True, auto_start=False
+        )
 
         # Daemon is not running, so should fallback to direct mode
         texts = ["def test():"]

